### PR TITLE
Provide coherent READMEs for each package.

### PR DESCRIPTION
**Pull request by peterbourgon**
RFCs were a good way to kickstart discussions, but I think we have enough well-formed opinions, and we're far enough along in implementation, that they're no longer necessary to corral discussion. This PR will give each package a succinct README that describes its _raison d'être_, rationale (drawn from RFCs where appropriate) and several usage examples. After that's done, I'll probably archive the RFCs and move the API stability policy to a docs/ folder. Comments and feedback welcome.

### CODE REVIEW

File: log/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# package log
+
+`package log` is an opinionated package for logging in your service.
+It provides a minimal interface for structured logging, which may be wrapped to encode conventions, enforce type-safety, etc.
+It can used for both typical application log events, and log-structured data streams.
```
Comment (ChrisHines) [inline log/README.md]: It can ^be^ used ...

**Comment by ChrisHines** [comment]
LGTM. We should work to put the code examples in the godoc of each package also.

**Comment by peterbourgon** [comment]
Yeah, I like the idea of proper examples. I'll roll that into this PR.

File: log/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# package log
+
+`package log` is an opinionated package for logging in your service.
```
Comment (harlow) [inline log/README.md]: Thoughts on removing:
> is an opinionated package for logging in your service.

Consider:

`package log` provides a minimal interface for structured logging, which may be wrapped to encode conventions, enforce type-safety, etc.

**Comment by harlow** [comment]
> Yeah, I like the idea of proper examples. I'll roll that into this PR.

Agreed :+1: 

Though I'd consider removing all the references to "it's an opinionated..." I feel like it's distracting to have in each of the intro paragraphs.

**Comment by peterbourgon** [comment]
@harlow if I merge this to master, would you be OK with rebasing your branch and submitting a new PR? Is that the easiest way?

**Comment by harlow** [comment]
@peterbourgon sure thing!

**Comment by peterbourgon** [comment]
Gonna continue with the rationales and proper examples in a subsequent PR by @harlow

**Comment by eduncan911** [comment]
Just a comment... What about auto generating the READMEs based on the go documentation of code?

http://godoc.org/github.com/davecheney/godoc2md

Then write an accompanying doc.go for each package. 

Output looks like this:

https://github.com/eduncan911/es
^- and peak through the importable packages (folders) for additional READMEs.

By having the `doc.go` file for each package and sub-package, not only is the github's `README.md` check, but `godoc.or/` is also kept in check with the exact same contents:

http://godoc.org/github.com/eduncan911/es

**Comment by eduncan911** [comment]
As for Travis build badges and other misc badge-links, put them in a BADGES.md file and write a small script that:

Loops through each directory.  If there is `*.go` code:

* 1) execute `godoc2md github.com/eduncan911/es/bus > README.md`
* 2) prepend: `cat BADGES.md README.md > README.md~; mv README.md~ README.md`

**Comment by ChrisHines** [comment]
I like the idea in general, but I'm not a fan of having the full API in the README. Is there any way to only include the package summary that godoc puts before the index, including package scoped examples?  That feels right to me. If someone wants to see all the docs a link to godoc.org is sufficient IMHO.

**Comment by eduncan911** [comment]
I believe that can be done with the templates, yes.  His needs work anyhow.

**Comment by peterbourgon** [comment]
@ChrisHines echoes my opinion exactly. I think the workflow is a good idea for the various packages. We can keep the badges exclusively in the top-level README, which I guess will always be handcrafted. I've created #52.

**Pull request merged**